app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/CachedTimelineRemoteMediator.kt

```diff
@@ -160,8 +161,9 @@ class CachedTimelineRemoteMediator(
                 if (oldStatus != null) break
             }
 
+            val filter = status.getApplicableFilter(viewModel.kind.toFilterKind())
             val expanded = oldStatus?.expanded ?: activeAccount.alwaysOpenSpoiler
-            val contentShowing = oldStatus?.contentShowing ?: (activeAccount.alwaysShowSensitiveMedia || !status.actionableStatus.sensitive)
+            val contentShowing = oldStatus?.contentShowing ?: (activeAccount.alwaysShowSensitiveMedia || (!status.actionableStatus.sensitive && filter?.action != Filter.Action.BLUR))
```

Comment:
This logic is repeated in a bunch of places, but I didn't find a nice way to extract it 🤷

Reply:
Yeah, I had exactly that in one iteration and couldn't decide if it was better or not - if you think it's an improvement, then I'll readd it